server/repository/src/migrations/v2_01_00/store_add_name_link_id.rs

```diff
@@ -0,0 +1,116 @@
+use crate::{migrations::sql, StorageConnection};
+
+pub(crate) fn migrate(connection: &StorageConnection) -> anyhow::Result<()> {
+    sql!(
+        connection,
+        r#"
+        DROP VIEW report_store;
+    "#,
+    )?;
+
+    #[cfg(feature = "postgres")]
+    sql!(
+        connection,
+        r#"
+        -- Adding store.name_link_id
+        DROP INDEX index_store_name_id_fkey;
+        ALTER TABLE store
+        ADD COLUMN name_link_id TEXT;
+        
+        UPDATE store
+        SET name_link_id = name_id;
+
+        ALTER TABLE store ADD CONSTRAINT store_name_link_id_fkey FOREIGN KEY (name_link_id) REFERENCES name_link(id);
+        ALTER TABLE store DROP COLUMN name_id;
+        "#,
+    )?;
+
+    #[cfg(not(feature = "postgres"))]
+    sql!(
+        connection,
+        r#"
+        -- Drop views early
+        DROP VIEW stock_on_hand;
+        DROP VIEW store_items;
+        -- consumption is recreated later in this migration
+        DROP VIEW IF EXISTS consumption;
+
+        PRAGMA foreign_keys=off;
+
+        CREATE TABLE store_new (
+          id TEXT NOT NULL PRIMARY KEY,
+          name_link_id TEXT NOT NULL REFERENCES name_link(id),
+          code TEXT NOT NULL,
+          site_id INTEGER NOT NULL,
+          logo TEXT,
+          store_mode TEXT DEFAULT 'STORE' NOT NULL,
+          created_date TEXT
+        );
+
+        INSERT INTO store_new SELECT * FROM store;
+
+        DROP TABLE store;
```

Comment:
Nit `PRAGMA foreign_keys=off;` can be moved just before `DROP TABLE store;` right? This makes it more clear where it is needed

Reply:
The PRAGMA is around the wrong statements, isn't it? Shouldn't it be around:

```
DROP TABLE store;
ALTER TABLE store_new RENAME TO store;
```